v0/destinations/vero/transform.js

```diff
@@ -0,0 +1,124 @@
+const set = require("set-value");
+const get = require("get-value");
+const {
+  defaultRequestConfig,
+  CustomError,
+  constructPayload,
+  getValueFromMessage,
+  isDefinedAndNotNull,
+  removeUndefinedAndNullAndEmptyValues
+} = require("../../util");
+const { EventType } = require("../../../constants");
+const { CONFIG_CATEGORIES, MAPPING_CONFIG } = require("./config");
+
+// This function handles the common response payload for the supported calls
+const commonResponseBuilder = (message, category, destination) => {
+  const payload = constructPayload(message, MAPPING_CONFIG[category.name]);
+  const authToken = get(destination, "Config.authToken");
+  set(payload, "auth_token", authToken);
+
+  // channels is defined only for Identify call
+  if (category.type === EventType.IDENTIFY) {
+    const channels = get(payload, "channels");
+    const address = get(channels, "address");
+    const platform = get(channels, "platform");
+
+    // Vero only accepts a valid channels object - address, platform and type should be defined and not null
+    if (isDefinedAndNotNull(address) && isDefinedAndNotNull(platform)) {
+      set(payload, "channels.type", category.channelType);
+    }
+    // Added this block because channels & channels.device can be defined and not null, which should be removed
+    else if (isDefinedAndNotNull(channels)) {
+      delete payload.channels;
+    }
+  }
+
+  const response = defaultRequestConfig();
+  response.endpoint = category.endpoint;
+  response.body.JSON = removeUndefinedAndNullAndEmptyValues(payload);
+  response.method = category.method;
+
+  return response;
+};
+
+// This function handles identify and track requests as well as addition-removal of tags.
+const identifyAndTrackResponseBuilder = (message, category, destination) => {
+  const respList = [];
+  const response = commonResponseBuilder(message, category, destination);
+  respList.push(response);
+
+  // This block handles any tag addition or removal requests.
+  // Ref - https://developers.getvero.com/?bash#tags
+  const tags = getValueFromMessage(message, "integrations.Vero.tags");
+  const addTags = get(tags, "add");
+  const removeTags = get(tags, "remove");
+  // Both add and remove should be an array and at least one should have a length > 0
+  if (
+    (Array.isArray(removeTags) && removeTags.length > 0) ||
+    (Array.isArray(addTags) && addTags.length > 0)
+  ) {
```

Comment:
For fetching values from `integrations` object this function from common untils `getIntegrationsObj(message, "destination_cname")` should be used
https://github.com/rudderlabs/rudder-transformer/blob/1537d3ec5d932e603a011a33c0fc53a43623b33d/v0/util/index.js#L826 also
https://github.com/rudderlabs/rudder-transformer/blob/master/constants/destinationCanonicalNames.js
should be populated

Reply:
Done